Assets/MixedRealityToolkit/Utilities/AssetScriptReferenceRetargeter.cs

```diff
@@ -0,0 +1,593 @@
+                        foundNonYamlExtensions.Add(extension);
+                    }
+
+                    bool copyFile = true;
+                    foreach (var suffix in ExcludedSuffixFromCopy)
+                    {
+                        if (filePath.EndsWith(suffix))
+                        {
+                            copyFile = false;
+                            break;
+                        }
+                    }
+
+                    if (copyFile)
+                    {
+                        File.Copy(filePath, targetPath);
+                    }
+                }
+            }
+
+            foreach (var extension in foundNonYamlExtensions)
+            {
+                Debug.Log($"Not a YAML extension: {extension}");
+            }
+
+            var tasks = yamlAssets.Select(t => Task.Run(() => ProcessYamlFile(t.Item1, t.Item2, remapDictionary)));
+            Task.WhenAll(tasks).Wait();
+
+            PostProcess(outputDirectory, dllGuids);
+        }
+
+        private static async Task ProcessYamlFile(string filePath, string targetPath, Dictionary<string, Tuple<string, long>> remapDictionary)
+        {
+            int lineNum = 0;
+            using (StreamReader reader = new StreamReader(filePath))
+            using (StreamWriter writer = new StreamWriter(targetPath))
+            {
+                while (!reader.EndOfStream)
+                {
+                    string line = await reader.ReadLineAsync();
+                    lineNum++;
+                    if (line.Contains("m_Script"))
+                    {
+                        if (!line.Contains('}'))
+                        {
+                            // Read the second line as well
+                            line += await reader.ReadLineAsync();
+                            lineNum++;
+
+                            if (!line.Contains('}'))
+                            {
+                                throw new InvalidDataException($"Unexpected part of YAML line split over more than two lines, starting two lines: {line}");
+                            }
+                        }
+
+                        if (line.Contains(ScriptFileIdConstant))
+                        {
+                            Match regexResults = Regex.Match(line, @"guid:\s*([0-9a-fA-F]*)");
+                            if (!regexResults.Success || regexResults.Groups.Count != 2 || !regexResults.Groups[1].Success || regexResults.Groups[1].Captures.Count != 1)
+                            {
+                                throw new InvalidDataException($"Failed to find the guid in line: {line}.");
+                            }
+
+                            string guid = regexResults.Groups[1].Captures[0].Value;
+                            if (remapDictionary.TryGetValue(guid, out Tuple<string, long> tuple))
+                            {
+                                line = $"  m_Script: {{fileID: {tuple.Item2}, guid: {tuple.Item1}, type: 3}}";
```

Comment:
>{ [](start = 53, length = 1)

Question for my own understanding - is this extra '{{' character designed to trigger some string replacement behavior or something? I see that in the YAML files that are checked in they're just like:

 m_Script: {fileID: 11500000, guid: e534d3d8fbada634785d567d8fe8562c, type: 3}

Instead of 

 m_Script: {{fileID: 11500000, guid: e534d3d8fbada634785d567d8fe8562c, type: 3}}

Reply:
It is the escape string for string format done through $. The actual string representation is a single '{'.

---
In reply to: [277833931](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/4051#discussion_r277833931) [](ancestors = 277833931)